power-up.js

```diff
@@ -45,4 +47,74 @@ class PowerUp extends Entity {
   static getRandomInt(max) {
     return Math.floor(Math.random() * Math.floor(max));
   }
+
+  shouldDrop() {
+    return PowerUp.getRandomInt(MAX_RANDOM) < this.dropRate;
+  }
+}
+
+class ExtraLife extends PowerUp {
+  constructor() {
+    super(true, 50);
+
+    this.manifest = {
+      owner: null,
+      angle: Math.PI / 2,
+      payload: {
+        type: {
+          sprite: sprite.rainbowBall,
+          radius: 30,
+        },
+        speed: 60,
+        powerUp() {
+          this.lives += 1;
+          addLife();
+        },
+      },
+    };
+  }
+}
+
+class Shield extends PowerUp {
+  constructor() {
+    super(false, 100);
+
+    this.manifest = {
+      owner: null,
+      angle: Math.PI / 2,
+      payload: {
+        type: {
+          sprite: sprite.shield,
+          radius: 30,
+        },
+        speed: 60,
+        powerUp() {
+          // Add the power up to the screen inventory
```

Comment:
For a shield what I imagined is we get access to the player's inventory and then add a function that resembles the powerups player modifications

In this case add a shield entity that takes a hit for the player